ptvsd/wrapper.py

```diff
@@ -545,12 +546,27 @@ def on_configurationDone(self, request, args):
         self.send_process_event(self.start_reason)
     
     def process_launch_arguments(self):
-        """Process the launch arguments to configure the debugger"""
+        """
+        Process the launch arguments to configure the debugger.
+        Further information can be found here https://code.visualstudio.com/docs/editor/debugging#_launchjson-attributes
+        {
+            type:'python',
+            request:'launch'|'attach',
+            name:'friendly name for debug config',
+            // Custom attributes supported by PTVSD.
+            redirectStdout:true,
+            redirectStderr:true,
+            fixFilePathCase:false
```

Comment:
We might as well do this always, or depending on the platform, since there's no reason why a protocol client would prefer lowercased path (if they are doing case-insensitive comparisons, they just need to do it on their end).

Reply:
> We might as well do this always, or depending on the platform

We don't need this for VS, this is a problem in VS Code. 

> since there's no reason why a protocol client would prefer lowercased path

Agreed, however on Windows, file paths are meant to be case insensitive. I'd rather have this a feature that can be used on demand.

Reply:
A Windows client that needs case-insensitive paths should use case-insensitive comparisons for the case-preserved paths that it receives, rather than relying on lowercased paths to begin with. VS is already doing that, so I wouldn't expect it to be negatively affected by paths not being lowercased.

On the other hand, lowercasing will sometimes be visible to the user even in VS, if not quite as annoying as in VSC. For example, suppose that you have a breakpoint or an exception inside some file that is *not* opened in VS. It will open it when it receives the stop event - but the file name and path in the tab will use the path as reported by debugger, rather than the proper case of the file on disk.

Reply:
> Agreed, however on Windows, file paths are meant to be case insensitive

This is not strictly true. Windows is case-insensitive but case-preserving and that distinction is very important. Any time code ever messes with the case of a filename and expects to be able to access it after is either broken at worst and unnecessary at best.

It is also [very easy to tweak Windows to be case-sensitive](https://superuser.com/questions/266110/how-do-you-make-windows-7-fully-case-sensitive-with-respect-to-the-filesystem), and I've ran Windows systems like this for years. It aids interoperability greatly not doing this.